src/crafting.cpp

```diff
@@ -2124,7 +2124,7 @@ item_location Character::create_in_progress_disassembly( item_location target )
 
     item_location disassembly_in_world;
 
-    if( !has_two_arms() ) {
+    if( !has_two_arms() || is_npc() ) {
```

Comment:
With #50479 disassembly will have a penalty for doing it on the ground, so it's better for NPCs to put the in-progress disassembly on the ground as a last resort.